cli/azd/pkg/config/config.go

```diff
@@ -56,6 +64,45 @@ func (c *config) Raw() map[string]any {
 	return c.data
 }
 
+// SetSecret stores the secrets at the specified path within a local user vault
+func (c *config) SetSecret(path string, value string) error {
+	if c.vaultId == "" {
+		c.vault = NewConfig(nil)
+		c.vaultId = uuid.New().String()
+		if err := c.Set("vault", c.vaultId); err != nil {
+			return fmt.Errorf("failed setting vault id: %w", err)
+		}
+	}
+
+	pathId := uuid.New().String()
+	vaultRef := fmt.Sprintf("vault://%s/%s", c.vaultId, pathId)
+	if err := c.vault.Set(pathId, base64.StdEncoding.EncodeToString([]byte(value))); err != nil {
```

Comment:
Just wondering - is the base64 encoding here just so we don't store the literal plaintext in the config file, even though it isn't encrypted (a-la a k8s `secret`) or is there some functional reason we needed to base64 encode this?

Reply:
Yes, just so we don't see the plain text... not mean to be any protection whatsoever.  This is similar to how K8S handles secrets.  They are all stored internally as base64 just so they are not in obvious plain text.